redfish-core/lib/roles.hpp

```diff
@@ -64,13 +68,57 @@ inline bool getAssignedPrivFromRole(std::string_view role,
     {
         privArray = nlohmann::json::array();
     }
+    else if (role == "OemIBMServiceAgent")
+    {
+        privArray = {"Login", "ConfigureManager", "ConfigureUsers",
+                     "ConfigureSelf", "ConfigureComponents"};
```

Comment:
Is there not an additional priv to add here (OemIBMPerformService)?
Please clarify if I'm misunderstanding.

Output from get role oemIBMServiceAgent:
curl -k -H "X-Auth-Token: $token" -X GET https://${bmc}/redfish/v1/AccountService/Roles/OemIBMServiceAget
{
  "@odata.id": "/redfish/v1/AccountService/Roles/OemIBMServiceAgent",
  "@odata.type": "#Role.v1_2_2.Role",
  _"AssignedPrivileges": [
    "Login",
    "ConfigureManager",
    "ConfigureUsers",
    "ConfigureSelf",
    "ConfigureComponents"
  ],_
  "Description": "OemIBMServiceAgent User Role",
  "Id": "OemIBMServiceAgent",
  "IsPredefined": true,
  "Name": "User Role",
  "OemPrivileges": [
    "OemIBMPerformService"
  ],
  "Restricted": true,
  "RoleId": "OemIBMServiceAgent"

Reply:
Yes, that is intended.  Redfish separates Redfish standard privileges and OEM privileges.